airflow/operators/mssql_to_hive.py

```diff
@@ -98,38 +99,34 @@ def __init__(
 
     @classmethod
     def type_map(cls, mssql_type):
-        t = pymssql
-        d = {
-            t.BINARY.value: 'INT',
-            t.DECIMAL.value: 'FLOAT',
-            t.NUMBER.value: 'INT',
+        map_dict = {
+            pymssql.BINARY.value: 'INT',
+            pymssql.DECIMAL.value: 'FLOAT',
+            pymssql.NUMBER.value: 'INT',
         }
-        return d[mssql_type] if mssql_type in d else 'STRING'
+        return map_dict[mssql_type] if mssql_type in map_dict else 'STRING'
 
     def execute(self, context):
-        hive = HiveCliHook(hive_cli_conn_id=self.hive_cli_conn_id)
         mssql = MsSqlHook(mssql_conn_id=self.mssql_conn_id)
-
         self.log.info("Dumping Microsoft SQL Server query results to local file")
-        conn = mssql.get_conn()
-        cursor = conn.cursor()
-        cursor.execute(self.sql)
-        with NamedTemporaryFile("w") as f:
-            csv_writer = csv.writer(f, delimiter=self.delimiter, encoding='utf-8')
-            field_dict = OrderedDict()
-            col_count = 0
-            for field in cursor.description:
-                col_count += 1
-                col_position = "Column{position}".format(position=col_count)
-                field_dict[col_position if field[0] == '' else field[0]] \
-                    = self.type_map(field[1])
-            csv_writer.writerows(cursor)
-            f.flush()
-            cursor.close()
-            conn.close()
+        with closing(mssql.get_conn()) as conn:
```

Comment:
using `with` will invoke the context library and call the __enter__ and __exit, is there a reason you are using closing again? Is it because mssql doesn't implement  those two functions?

Reply:
2nd: have you checked whether `with mssql.get_conn()` will return a cursor or not? From my experience, I have seen couples of them do. If that's the case, having a second with loop will encounter NPE.

Reply:
Thanks @feng-tao for noticing and reporting this to me :) I really haven't thought about that.

I found an example from the official pymssql docs: 
http://pymssql.org/en/stable/pymssql_examples.html#using-the-with-statement-context-managers

So you are right. `closing` is not necessary here. I will fix this :)

Reply:
The tests now correctly check if `__enter__` gets called